ZenPacks/zenoss/LinuxMonitor/modeler/plugins/zenoss/cmd/linux/interfaces.py

```diff
@@ -198,7 +198,12 @@ def parseIfconfig(self, ifconfig, device, relMap):
                     ip, netmask = maddr.groups()[:2]
                 else:
                     ip, netmask = maddr.groups()[2:]
-                netmask = self.maskToBits(netmask)
+                if '.' in netmask:
+                    netmask = self.maskToBits(netmask)
+                else:
+                    if not netmask.startswith('0x'):
+                        '0x' + netmask
```

Comment:
This statement has no effect. Perhaps you meant `netmask = '0x' + netmask`?

This still looks like it has a lot of failure modes. maskToBits can still fail even if a . appears somewhere in netmask, and hexToBits seems like it returns 32 if you feed it garbage. Something like the following would be safer.


```python

import re
from Products.ZenUtils.IpUtil import isip

###

if isip(netmask):
    netmask = self.maskToBits(netmask)
elif re.match(r'([0-9a-fA-F]{8}', netmask):
    netmask = self.hexToBits("0x" + netmask)
else:
    # Assume a /32 if netmask can't be found.
    netmask = 32
```